lib/smartdown/model/node.rb

```diff
@@ -4,7 +4,14 @@ module Smartdown
   module Model
     Node = Struct.new(:name, :elements, :front_matter) do
       def initialize(name, elements, front_matter = nil)
-        super(name, elements, front_matter || Smartdown::Model::FrontMatter.new)
+        all_elements = elements.map do |element|
```

Comment:
I'd be good to have an inline comment here (and/or in the `Elements` class itself) explaining why there is the `Elements` model that needs flattening. We don't want too much information about how the parser works to leak into other layers, but we should explain it somewhere. What do you think?